ddl/partition.go

```diff
@@ -365,6 +377,52 @@ func onDropTablePartition(t *meta.Meta, job *model.Job) (ver int64, _ error) {
 	return ver, nil
 }
 
+// onDropTablePartition truncates old partition meta.
+func onTruncateTablePartition(t *meta.Meta, job *model.Job) (int64, error) {
+	var ver int64
+	var oldID int64
+	if err := job.DecodeArgs(&oldID); err != nil {
+		job.State = model.JobStateCancelled
+		return ver, errors.Trace(err)
+	}
+	tblInfo, err := getTableInfo(t, job, job.SchemaID)
+	if err != nil {
+		return ver, errors.Trace(err)
+	}
+	pi := tblInfo.GetPartitionInfo()
+	if pi == nil {
+		return ver, errors.Trace(ErrPartitionMgmtOnNonpartitioned)
+	}
+
+	var find bool
+	for i := 0; i < len(pi.Definitions); i++ {
+		def := &pi.Definitions[i]
+		if def.ID == oldID {
+			pid, err1 := t.GenGlobalID()
+			if err != nil {
+				return ver, errors.Trace(err1)
+			}
+			def.ID = pid
+			find = true
+			break
+		}
+	}
+	if !find {
+		return ver, errUnknownPartition.GenWithStackByArgs("drop?", tblInfo.Name.O)
```

Comment:
The error msg is different with mysql, right?

Reply:
This is the advice from @winkyao , after changing the `job.Args` to id, there is no partition name available here.
MySQL never face problem that many concurrency connection modifying the DDL job queue. @crazycs520